src/GitHub.VisualStudio/Menus/CopyLink.cs

```diff
@@ -45,7 +48,11 @@ public void Activate([AllowNull]object data = null)
 
         public bool CanShow()
         {
-            return IsGitHubRepo();
+            return System.Threading.Tasks.Task.Run(() =>
+            {
+                return IsGitHubRepo(); 
+            }).Result;
```

Comment:
This will block the UI for as much time as it takes the task to run, which means it will block the context menu from showing up for that amount of time. You only want to call `Result` on a task when you know the task is done, otherwise there's a risk of deadlocks and visible blocking. You also really don't need to run a task here, `IsGitHubRepo` is already a task.

The best way to do this is to call `Wait` on the task returned by `IsGitHubRepo`, with a time of around 250 milliseconds, which is a delay low enough to not annoy users when they right click, but high enough that if there's no need for internet access, it will complete under that time (for github.com repos).

If `Wait` returns true, then the task is done and you can return the `Result` safely, otherwise it's not done yet and the 250 milliseconds have already elapsed so we don't know if it's one of our repos or not. In this second case we just return false (in case it's not one of our repos), and the next time we evaluate it, `SimpleApi` will have completed and cached the results, so there won't be any delay.